In [1]:
import pandas as pd
import numpy as np
import pyreadstat

In [2]:
spss_2018_df, meta_2018 = pyreadstat.read_sav('./data/2018방송매체이용행태조사_가구_최종.sav')
spss2_2018_df, meta2_2018 = pyreadstat.read_sav('./data/2018방송매체이용행태조사_개인_최종.sav')

In [3]:
raw_home_2018_df = pd.read_spss('./data/2018방송매체이용행태조사_가구_최종.sav')
raw_personal_2018_df = pd.read_spss('./data/2018방송매체이용행태조사_개인_최종.sav')

온라인동영상제공서비스 : OTT
- 381 부터 OTT
- p_r001_m01 : 평소에 이용한 온라인동영상제공서비스(OTT)


In [4]:
print(meta2_2018.column_names_to_labels)

{'pid': 'Panel ID', 'weight': '가중치', 't_mn1': '가구원 수', 'region': '시부/군부', 'fac': '가족구성', 'income1': '월 평균 가구소득', 'income2': '가구소득', 'age2': '연령', 'p_reg2': '지역', 'p_sex': '성별', 'p_school1': '학력', 'p_occ2': '직업', 'p_res2': '주거형태', 'p_city_g': '도시규모', 'TV': 'TV 소유', 'RD': '라디오 소유', 'CT': '컴퓨터/테블릿PC 소유', 'DM': '차량용 DMB 단말기 소유', 'DV': 'VCR/DVD플레이어/블루레이 플레이어 소유', 'DC': '디지털 카메라/캠코더 소유', 'GA': '가구 보유형 게임기 소유', 'ST': '스마트형 TV 수신 장치 소유', 'TE': '가정용 전화 회선 사용', 'p_ba_tv': '지상파TV만 이용', 'p_ca_tv': '케이블 방송', 'p_sky': '위성방송', 'p_iptv': 'IPTV', 'TV_1': '지상파만 이용-리코딩', 'TV_2': '케이블 방송 이용-리코딩', 'TV_3': '위성방송 이용-리코딩', 'p_iptv_r': 'IPTV 이용-리코딩', 'TV_5': '아날로그 케이블 이용-리코딩', 'TV_6': '디지털 케이블 이용-리코딩', 'DM10_1': '아날로그 케이블', 'DM10_2': '디지털 케이블', 'p_bDC003': '1-6 스마트워치(대).소유대수', 'p_bZZ003': '1-6 스마트 워치 소유', 'p_bMP001': '1-3 MP3플레이어 소유', 'p_bMP002': '1-3 MP3(대).플레이어 소유 대수', 'p_bMP003': '1-3 라디오(대).라디오수신가능 MP3', 'p_bCE001': '1-1 휴대폰 소유', 'p_bCE002': '1-1 총(대).휴대폰 총 소유 대수', 'p_bCE003': '1-1 일반폰(대).소유 대수', 'p_bCE005

In [4]:
raw_personal_2018_df.head(3)

pid    weight     t_mn1 region           fac       income1    income2  \
0  101.0  0.364503     2인 가구     군부     1세대가구(부부)  200-250만원 미만  200-299만원   
1  102.0  0.651482     2인 가구     군부     1세대가구(부부)  200-250만원 미만  200-299만원   
2  201.0  0.800944  3인 이상 가구     군부  2세대가구(부부+자녀)  300-350만원 미만  300-399만원   

  age2 p_reg2 p_sex  ... p_r008_m1 p_r008_m2 p_r008_m3 p_r008_m4 p_r008_m5  \
0  50대  대구/경북    여자  ...       NaN       NaN       NaN       NaN       NaN   
1  60대  대구/경북    남자  ...       NaN       NaN       NaN       NaN       NaN   
2  60대  대구/경북    남자  ...       NaN       NaN       NaN       NaN       NaN   

  p_r011_m1 p_r011_m2 p_r011_m3 p_r011_m4 p_r011_m5  
0       NaN       NaN       NaN       NaN       NaN  
1       NaN       NaN       NaN       NaN       NaN  
2       NaN       NaN       NaN       NaN       NaN  

[3 rows x 440 columns]

# 2018방송매체이용행태조사_가구&개인 데이터 확인


* home_2018_df: 가구 데이터

* personal_2018_df: 개인 데이터

In [5]:
home_2018_df = raw_home_2018_df.copy()
personal_2018_df = raw_personal_2018_df.copy()

In [6]:
personal_label_2018 = meta2_2018.column_names_to_labels
personal_2018_df.rename(columns=personal_label_2018, inplace=True)

In [7]:
home_label_2018 = meta_2018.column_names_to_labels
home_2018_df.rename(columns=home_label_2018, inplace=True)

In [8]:
personal_2018_df.head()

Panel ID       가중치     가구원 수 시부/군부          가족구성     월 평균 가구소득       가구소득  \
0     101.0  0.364503     2인 가구    군부     1세대가구(부부)  200-250만원 미만  200-299만원   
1     102.0  0.651482     2인 가구    군부     1세대가구(부부)  200-250만원 미만  200-299만원   
2     201.0  0.800944  3인 이상 가구    군부  2세대가구(부부+자녀)  300-350만원 미만  300-399만원   
3     202.0  0.448127  3인 이상 가구    군부  2세대가구(부부+자녀)  300-350만원 미만  300-399만원   
4     203.0  1.623214  3인 이상 가구    군부  2세대가구(부부+자녀)  300-350만원 미만  300-399만원   

    연령     지역  성별  ... 문16.온라인 동영상 서비스로 이용한 시청 프로그램 유형  \
0  50대  대구/경북  여자  ...                             NaN   
1  60대  대구/경북  남자  ...                             NaN   
2  60대  대구/경북  남자  ...                             NaN   
3  50대  대구/경북  여자  ...                             NaN   
4  30대  대구/경북  남자  ...                             NaN   

  문16.온라인 동영상 서비스로 이용한 시청 프로그램 유형 문16.온라인 동영상 서비스로 이용한 시청 프로그램 유형  \
0                             NaN                             NaN   
1                             NaN                             NaN   
2                             NaN                             NaN   
3                             NaN                             NaN   
4                             NaN                             NaN   

  문16.온라인 동영상 서비스로 이용한 시청 프로그램 유형 문16.온라인 동영상 서비스로 이용한 시청 프로그램 유형  \
0                             NaN                             NaN   
1                             NaN                             NaN   
2                             NaN                             NaN   
3                             NaN                             NaN   
4                             NaN                             NaN   

  문12.온라인동영상제공서비스 이용 기기 문12.온라인동영상제공서비스 이용 기기 문12.온라인동영상제공서비스 이용 기기  \
0                   NaN                   NaN                   NaN   
1                   NaN                   NaN                   NaN   
2                   NaN                   NaN                   NaN   
3                   NaN                   NaN                   NaN   
4                   NaN                   NaN                   NaN   

  문12.온라인동영상제공서비스 이용 기기 문12.온라인동영상제공서비스 이용 기기  
0                   NaN                   NaN  
1                   NaN                   NaN  
2                   NaN                   NaN  
3                   NaN                   NaN  
4                   NaN                   NaN  

[5 rows x 440 columns]

In [10]:
# personal_2018_df[personal_2018_df['문16.온라인 동영상 서비스로 이용한 시청 프로그램 유형'].isna() != True]

In [9]:
home_2018_df.head()

Panel ID       가중치     가구원 수 시부/군부  총 가구원 수(명)          가족구성     월 평균 가구소득  \
0       1.0  0.561146     2인 가구    군부         2.0     1세대가구(부부)  200-250만원 미만   
1       2.0  0.689883  3인 이상 가구    군부         3.0  2세대가구(부부+자녀)  300-350만원 미만   
2       3.0  0.561146     2인 가구    군부         2.0     1세대가구(부부)   50-100만원 미만   
3       4.0  0.689883  3인 이상 가구    군부         3.0  2세대가구(부부+자녀)  250-300만원 미만   
4       5.0  0.561146     2인 가구    군부         2.0     1세대가구(부부)  150-200만원 미만   

        가구소득  주거형태 TV 소유  ...  케이블 방송 이용  위성방송 이용  IPTV 이용  지상파TV만 이용 케이블 방송  \
0  200-299만원  단독주택    소유  ...        비이용       이용       이용        비이용    비이용   
1  300-399만원  단독주택    소유  ...        비이용       이용       이용        비이용    비이용   
2   100만원 미만  단독주택    소유  ...         이용      비이용      비이용        비이용     이용   
3  200-299만원  단독주택    소유  ...         이용      비이용      비이용        비이용     이용   
4  100-199만원  단독주택    소유  ...        비이용       이용       이용        비이용    비이용   

   위성방송  IPTV  아날로그 케이블 디지털 케이블     지역  
0    이용    이용       비이용     비이용  대구/경북  
1    이용    이용       비이용     비이용  대구/경북  
2   비이용   비이용       비이용      이용  대구/경북  
3   비이용   비이용       비이용      이용  대구/경북  
4    이용    이용       비이용     비이용  대구/경북  

[5 rows x 142 columns]

In [12]:
personal_2018_df['TV 평일 평균 이용시간(분 단위)(이용자)']

0       180.0
1        90.0
2       240.0
3       240.0
4        30.0
        ...  
7229    180.0
7230      NaN
7231    120.0
7232    120.0
7233    120.0
Name: TV 평일 평균 이용시간(분 단위)(이용자), Length: 7234, dtype: float64

In [13]:
personal_2018_df['문6-1.최근일주일 유료방송 시청 채널 유형']

문6-1.최근일주일 유료방송 시청 채널 유형 문6-1.최근일주일 유료방송 시청 채널 유형  \
0                     뉴스/시사보도                       교양   
1                     뉴스/시사보도                       교양   
2                     뉴스/시사보도                       예능   
3                     뉴스/시사보도                       교양   
4                     뉴스/시사보도                       교양   
...                       ...                      ...   
7229                  뉴스/시사보도                       교양   
7230                  시청하지 않음                      NaN   
7231                  뉴스/시사보도                       영화   
7232                      드라마                       예능   
7233                  뉴스/시사보도                       교양   

     문6-1.최근일주일 유료방송 시청 채널 유형 문6-1.최근일주일 유료방송 시청 채널 유형  \
0                         드라마                       예능   
1                         NaN                      NaN   
2                         스포츠                      NaN   
3                         드라마                       예능   
4                         드라마                       예능   
...                       ...                      ...   
7229                      NaN                      NaN   
7230                      NaN                      NaN   
7231                      NaN                      NaN   
7232                      NaN                      NaN   
7233                      드라마                       예능   

     문6-1.최근일주일 유료방송 시청 채널 유형 문6-1.최근일주일 유료방송 시청 채널 유형  \
0                         스포츠                       영화   
1                         NaN                      NaN   
2                         NaN                      NaN   
3                         스포츠                      NaN   
4                         스포츠                       영화   
...                       ...                      ...   
7229                      NaN                      NaN   
7230                      NaN                      NaN   
7231                      NaN                      NaN   
7232                      NaN                      NaN   
7233                      홈쇼핑                      NaN   

     문6-1.최근일주일 유료방송 시청 채널 유형 문6-1.최근일주일 유료방송 시청 채널 유형  \
0                         NaN                      NaN   
1                         NaN                      NaN   
2                         NaN                      NaN   
3                         NaN                      NaN   
4                         NaN                      NaN   
...                       ...                      ...   
7229                      NaN                      NaN   
7230                      NaN                      NaN   
7231                      NaN                      NaN   
7232                      NaN                      NaN   
7233                      NaN                      NaN   

     문6-1.최근일주일 유료방송 시청 채널 유형 문6-1.최근일주일 유료방송 시청 채널 유형  \
0                         NaN                      NaN   
1                         NaN                      NaN   
2                         NaN                      NaN   
3                         NaN                      NaN   
4                         NaN                      NaN   
...                       ...                      ...   
7229                      NaN                      NaN   
7230                      NaN                      NaN   
7231                      NaN                      NaN   
7232                      NaN                      NaN   
7233                      NaN                      NaN   

     문6-1.최근일주일 유료방송 시청 채널 유형 문6-1.최근일주일 유료방송 시청 채널 유형  
0                         NaN                      NaN  
1                         NaN                      NaN  
2                         NaN                      NaN  
3                         NaN                      NaN  
4                         NaN                      NaN  
...                       ...                      ...  
7229                      NaN                      NaN  
7230                      NaN                      NaN  
7231   

데이터 해석예시

0번 응답자는 최근 일주일 유로 방송으로 교양, 드라마, 예능, 스포츠, 영화를 시청했다.

7229번 응답자는 최근 일주일 유로 방송으로 교양을 시청했다.

In [15]:
temp_df = raw_personal_2018_df[(raw_personal_2018_df['p_n004_24'] == '드라마') & (raw_personal_2018_df['p_n004_25'] == '뉴스/시사보도')]
temp_df[['p_n004_24', 'p_n004_25']]

p_n004_24 p_n004_25
3109       드라마   뉴스/시사보도

In [16]:
temp_df = raw_personal_2018_df[(raw_personal_2018_df['p_n004_24'] == '뉴스/시사보도') | (raw_personal_2018_df['p_n004_25'] == '뉴스/시사보도') | (raw_personal_2018_df['p_n004_26'] == '뉴스/시사보도')]
# temp_df[['p_n004_24', 'p_n004_25']]
temp_df.count()

pid          5320
weight       5320
t_mn1        5320
region       5320
fac          5320
             ... 
p_r011_m1    1366
p_r011_m2     116
p_r011_m3      19
p_r011_m4       1
p_r011_m5       0
Length: 440, dtype: int64

In [31]:
raw_personal_2018_df.iloc[:, 381:392]

p_r001_1 p_r001_m01 p_r001_m02 p_r001_m03 p_r001_m04 p_r001_m05  \
0           NaN    이용한적 없음        NaN        NaN        NaN        NaN   
1           NaN    이용한적 없음        NaN        NaN        NaN        NaN   
2           NaN    이용한적 없음        NaN        NaN        NaN        NaN   
3           NaN    이용한적 없음        NaN        NaN        NaN        NaN   
4           NaN    이용한적 없음        NaN        NaN        NaN        NaN   
...         ...        ...        ...        ...        ...        ...   
7229        NaN    이용한적 없음        NaN        NaN        NaN        NaN   
7230        NaN    이용한적 없음        NaN        NaN        NaN        NaN   
7231        NaN    이용한적 없음        NaN        NaN        NaN        NaN   
7232         매일        유튜브        NaN        NaN        NaN        NaN   
7233  일주일에 3-4일        유튜브        NaN        NaN        NaN        NaN   

     p_r001_m06 p_r001_m07 p_r001_m08 p_r001_m09 p_r001_m10  
0           NaN        NaN        NaN        NaN        NaN  
1           NaN        NaN        NaN        NaN        NaN  
2           NaN        NaN        NaN        NaN        NaN  
3           NaN        NaN        NaN        NaN        NaN  
4           NaN        NaN        NaN        NaN        NaN  
...         ...        ...        ...        ...        ...  
7229        NaN        NaN        NaN        NaN        NaN  
7230        NaN        NaN        NaN        NaN        NaN  
7231        NaN        NaN        NaN        NaN        NaN  
7232        NaN        NaN        NaN        NaN        NaN  
7233        NaN        NaN        NaN        NaN        NaN  

[7234 rows x 11 columns]

In [19]:
print(raw_personal_2018_df.iloc[:,[382]].value_counts())

p_r001_m01  
이용한적 없음         5094
유튜브             1124
페이스북             290
네이버 TV           271
Pooq              87
아프리카 TV           84
Oksusu            82
Olleh TV 모바일      39
유플러스 비디오포럼        34
Tving             32
카카오 TV            29
곰TV               19
기타                17
에브리온 TV            9
넷플릭스               8
판도라TV              8
V live             4
왓챠플레이              2
데일리 모션             1
dtype: int64


In [20]:
print(raw_personal_2018_df.iloc[:,[382]].count())

p_r001_m01    7234
dtype: int64
